apps/favorite_products/lib/favorite_products/nectar_extension.ex

```diff
@@ -0,0 +1,52 @@
+defmodule FavoriteProducts.NectarExtension do
+  defmacro __using__([install: install_type]) do
+    do_install(install_type)
+  end
+
+  defp do_install("router") do
+    quote do
+      define_route do
+        scope "/", FavoriteProducts do
+          ## Do not forget to add the pipelines request should go through
+          pipe_through [:browser, :browser_auth]
+          resources "/favorites", FavoriteController, only: [:index, :create, :delete]
+        end
+      end
+    end
+  end
+
+  defp do_install("products") do
+    quote do
+      ## In Phoenix App Model Schema definition, join association is defined first and then through association
+      ## Please note the reverse order here as while collecting, it gets the reverse order and
+      ## injected into actual model schema as expected .. tricky huhh !!
+      add_to_schema(:has_many, :liked_by, through: [:likes, :user])
+      add_to_schema(:has_many, :likes, FavoriteProducts.UserLike, [])
+      include_method do
```

Comment:
@mgwidmann, we can make it work as you suggested but prefer to keep it for reasons as mentioned below :blush: :+1: 

`include_method` wrapping is chosen to help with readability and intent of inclusion (support functions here from extensions)

As, we are targeting a framework where we can reason about any additional block and **able to communicate/converse with fellow team members**, we introduced a common term to base our discussions on :blush: 

As we will come across more use-cases of extensions, everything still will be a function helping us achieve the needed requirement / feature and the **scoping of methods under `include_method`, `create_event`, `extend_channel` will help us reason about them in a controlled / defined manner.**